src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -0,0 +1,338 @@
+            try {
+                final double height = extractHeight(params);
+                final double weight = extractWeight(params);
+                return new CalculateBmiCommand(height, weight);
+            } catch (ParamInvalidException e) {
+                return new InvalidCommand(e.getMessage());
+            }
+        } else {
+            return new InvalidCommand(CalculateBmiCommand.MESSAGE_INVALID_COMMAND_FORMAT);
+        }
+    }
+
+    private Command parseChangeHeight(String params) {
+        try {
+            final double height = Double.parseDouble(params);
+            return new ChangeHeightCommand(height);
+        } catch (NumberFormatException e) {
+            return new InvalidCommand(MESSAGE_ERROR_NOT_A_NUMBER);
+        }
+    }
+
+    private Command parseChangeWeight(String params) {
+        try {
+            final double weight = Double.parseDouble(params);
+            return new ChangeWeightCommand(weight);
+        } catch (NumberFormatException e) {
+            return new InvalidCommand(MESSAGE_ERROR_NOT_A_NUMBER);
+        }
+    }
+
+    private Command parseCreateProfile(String params) {
+        if (!hasRequiredParams(params,
+                Command.COMMAND_PREFIX_NAME,
+                Command.COMMAND_PREFIX_HEIGHT,
+                Command.COMMAND_PREFIX_WEIGHT)) {
+            return new InvalidCommand(CreateProfileCommand.MESSAGE_INVALID_COMMAND_FORMAT);
+        }
+        try {
+            final String name = extractProfileName(params);
+            final double height = extractHeight(params);
+            final double weight = extractWeight(params);
+            return new CreateProfileCommand(name, height, weight);
+        } catch (ParamInvalidException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    private Command parseSetGoal(String params) {
+        try {
+            final int goal = Integer.parseInt(params);
+            return new SetGoalCommand(goal);
+        } catch (NumberFormatException e) {
+            return new InvalidCommand(MESSAGE_ERROR_NOT_A_NUMBER);
+        }
+    }
+
+    /**
+     * Returns a String array where 0th index is command string and 1st index is the remaining parameters.
+     * Command string and parameter string is assumed to be separated by the first " " in input.
+     * If no parameters are provided in the input, 1st index will be set to EMPTY.
+     *
+     * @param input Raw user input string
+     * @return String array [command, parameters]
+     */
+    private String[] splitInputIntoCommandAndParams(String input) {
+        String[] commandAndParams = new String[2];
+        final String[] inputSplit = input.trim().split(" ", 2);
+        //command string
+        commandAndParams[0] = inputSplit[0];
+        //param string; if not given, set to EMPTY for error handling
+        commandAndParams[1] = (inputSplit.length >= 2) ? inputSplit[1].trim() : EMPTY;
```

Comment:
Perhaps == 2 instead of >= 2as the top only string only split the string into 2 parts

Reply:
Okie!